### PR TITLE
MM-57867 Don't delete existing channel on invitation error

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
This PR addresses an issue where sharing a channel can result in a channel being deleted if the invite errors.  Specifically,  A remote can send an invite with a channel ID that exists on the local server, then send that same invite again, which errors because of a constraint violation and permanently deletes the local channel.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-57867

#### Release Note
```release-note
Fixes issue where sharing a channel can result in a channel being deleted if the invite errors.
```

### CODE REVIEW

File: server/platform/services/sharedchannel/channelinvite.go [hunk 1]
```diff
@@ -162,14 +162,21 @@ func (scs *Service) onReceiveChannelInvite(msg model.RemoteClusterMsg, rc *model
...
-	channel, err := scs.server.GetStore().Channel().Get(invite.ChannelId, true)
-	if err != nil {
-		if channel, err = scs.handleChannelCreation(invite, rc); err != nil {
-			return err
-		}
+	// check if channel already exists
+	var channel *model.Channel
+	_, err := scs.server.GetStore().Channel().Get(invite.ChannelId, true)
+	if err == nil {
+		// the channel already exists on this server; could be the remote is trying to re-share it (not allowed at this time).
```
Comment (wiggin77) [inline server/platform/services/sharedchannel/channelinvite.go:169]: We decided to not allow re-sharing channels for now.  A new ticket has been created to send a notice to the sender that the invite failed.

File: server/platform/services/sharedchannel/channelinvite.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ func (scs *Service) SendChannelInvite(channel *model.Channel, userId string, rc
 		ChannelId:   channel.Id,
 		TeamId:      rc.RemoteTeamId,
 		ReadOnly:    sc.ReadOnly,
-		Name:        sc.ShareName,
+		Name:        channel.Name,
```
Comment (wiggin77) [inline server/platform/services/sharedchannel/channelinvite.go:57]: Un-related but fixes an issue that makes testing this fix more difficult.

**Comment by wiggin77** [comment]
update-branch

**Review by esarafianou — APPROVED** [review]

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

**Review by streamer45 — APPROVED** [review]

**Comment by amyblais** [comment]
@wiggin77 Would you be able to help with a manual cherry-pick PR for ``release-9.5`` branch?

**Comment by wiggin77** [comment]
> @wiggin77 Would you be able to help with a manual cherry-pick PR for `release-9.5` branch?

@amyblais I'm having trouble finding a PR that's failing.

**Comment by amyblais** [comment]
@wiggin77 This error got post when I attempted to open an automated cherry-pick PR https://github.com/mattermost/mattermost/pull/27143#issuecomment-2145339068

**Comment by wiggin77** [comment]
> @wiggin77 This error got posted when I attempted to open an automated cherry-pick PR [#27143 (comment)](https://github.com/mattermost/mattermost/pull/27143#issuecomment-2145339068)

Ah I see.  I thought it created a PR that was failing.  It doesn't create the PR at all if it fails.

**Comment by wiggin77** [comment]
> @wiggin77 This error got posted when I attempted to open an automated cherry-pick PR [#27143 (comment)](https://github.com/mattermost/mattermost/pull/27143#issuecomment-2145339068)

@amyblais the manual cherry-pick PR is here:  https://github.com/mattermost/mattermost/pull/27270

**Pull request merged**